controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -353,6 +360,17 @@ protected void startUp() {
                 grpcServer.awaitRunning();
             }
 
+            HealthContributor rootContrib = healthServiceManager.getRoot();
+            SegmentContainerMonitorHealthContributor segmentContainerMonitorHC = new SegmentContainerMonitorHealthContributor("segmentContainerMonitor", monitor );
```

Comment:
this needs to be conditionally created based on `serviceConfig.getHostMonitorConfig().isHostMonitorEnabled()`
Please verify other components as well on similar lines.